### DISCUSSION TITLE
"new script" button and the ability to alter the default formating, also TAB hopping.

### DISCUSSION THREAD

**Discussion by EggofTreachery** [chuangcaleb/obsidian-fountain-editor#37] [Ideas]
Howdy! Love the plug-in. Just wanted to ask if it was possible to add it in as a "new script" button instead of the css properties? I normally don't mess with that portion so it took me a bit to figure out how to turn the page into a "script page" lol. I think it would help new users to the plug-in/ obsidian in general coming from other programs but I also acknowledged that I may just be a little dumb haha.

Also, I'm moving over from Scrivner and was wondering if there would be a possibility of customizing the formatting in the settings of the plug-in in the future. for instance if I say, wanted names to be bold instead of just capitalized I could go into the setting and set up the custom formatting, same if I wanted @ to automatically capitalize the word/ words after it instead. Or hell, make names blue or something.

Another thing that could be interesting/ helpful for workflow is tab hopping setting. I used it in Scrivner a lot (not sure if that's the actual term for it but its what i called it) and essentially it just lets you press tab to go between different formats/syntax depending on what the previous syntax is. Like say after a header you press enter and the next scenes goes to its default -action. Well when you're there you can press TAB to change it from action to character. You can still use the regular syntax to mess with things it just auto switches for you. I think holding TAB let you choose and click a syntax too?

Regardless this editor rocks!!!! TY for making it!

**Comment by chuangcaleb** [comment]
Heyo! Thanks for the love 🙆‍♂️

A command like "Initialise Fountain styling in note" would be a good low-hanging fruit to get, I think we can do that. For casual Obsidian users, adding frontmatter properties is quite expectedly simple, but yeah I think that shouldn't be too tough to add.

You can change the styling of Fountain tokens! ...if you know some HTML/CSS... you can make any token any color. Regarding capitalising, there is `text-transform: capitalize` but that's purely cosmetic and the underlying text may still be lowercase, so I think this may require some logic hacking. There is a [Dynamic Highlights plugin](obsidian://show-plugin?id=obsidian-dynamic-highlights) that may interest you in regards to highlighting specific keywords.

I don't think I personally will implement tab hopping (as you described it), I don't see myself ever using it, because the syntax handles everything explicitly! But pull requests from the community are always welcome.

**Reply by EggofTreachery** [reply]
Oh cool thanks for the info on stylizing stuff!
